src/LazySet.jl

```diff
@@ -191,12 +191,50 @@ function an_element(S::LazySet{N}) where {N<:Real}
 end
 
 
-# catch-all definition for ==
-function ==(a::L, b::L) where L<:LazySet
-    for f in fieldnames(a)
-        if getfield(a, f) != getfield(b, f)
+"""
+    ==(a::LazySet, b::LazySet)
+
+Return whether two LazySets of the same type are exactly equal by recursively
+comparing their fields until a mismatch is found.
+
+### Input
+
+- `a` -- any `LazySet`
+- `b` -- another `LazySet` of the same type as `a`
+
+### Output
+
+- `true` iff `a` is equal to `b`.
+
+### Notes
+
+The check is purely syntactic and the sets need to have the same base type.
+I.e. `a::VPolytope == b::HPolytope` returns `false` even if `a` and `b` represent the
+same polytope. However `a::HPolytope{Int64} == b::HPolytope{Float64}` is a valid comparison.
+
+### Examples
+```jldoctest
+julia> HalfSpace([1], 1) == HalfSpace([1], 1)
+true
+
+julia> HalfSpace([1], 1) == HalfSpace([1.0], 1.0)
+true
+
+julia> Ball1([0.], 1.) == Ball2([0.], 1.)
+false
+```
+"""
+function ==(X::LazySet, Y::LazySet)
+    # if the common supertype of a and b is abstract, they cannot be compared
```

Comment:
Sorry, I renamed the variables, but not in the comment :disappointed:

Reply:
Ha! I hadn't noticed. If you don't mind I'll change them back to a and b to match the doc

Reply:
okay

Reply:
Really I was just being lazy 😞 
I prefer X and Y too honestly...
Renamed again.